ui/src/pages/namespace/Explorer/Page/poc/PageCompiler/context/utils.ts

```diff
@@ -98,3 +98,21 @@ export const pathsEqual = (a: PathOrNull, b: PathOrNull) => {
   }
   return a.length === b.length && a.every((val, index) => val === b[index]);
 };
+
+export const getPlaceholderBlock = (type: AllBlocksType["type"]) => {
+  switch (type) {
+    case "headline":
+      return {
+        type: "headline",
+        level: "h1",
+        label: "",
+      } as AllBlocksType;
+    case "text":
+      return {
+        type: "text",
+        content: "",
+      } as AllBlocksType;
```

Comment:
```suggestion
       case "headline":
      return {
        type: "headline",
        level: "h1",
        label: "",
      } satisfies HeadlineType;
    case "text":
      return {
        type: "text",
        content: "",
      } satisfies TextType;
 ```
Good idea to add a util for this. However, we should use `satisfies` instead of typecasting. This way, typescript will report an error if the JSON doesn't match the intended type. And, we should use the narrow types of each specific block type. These are also used in the editing components, making sure that we cannot load the wrong block type into a component.